src/wasm-binary.h

```diff
@@ -116,6 +116,7 @@ struct LEB {
   }
 };
 
+#define SIZE32LEB(v) U32LEB((uint32_t)(v))
```

Comment:
not sure I like a macro here.

maybe the bigger question is, do we need this cast? and the next cast after it, `int32_t ret = (int32_t)o.size();` - are those strictly necessary? they add code clutter and confusion. i'm not sure i understand why we need them.